File: myurlshortner-consumer/myurlshortner-consumer/src/main/scala/com/acme/myurlshortner/consumer/domain/service/ShortenedUrlUserEventsService.scala
package com.acme.myurlshortner.consumer.domain.service

import com.acme.myurlshortner.consumer.domain.command.SaveShortenedUrlUserAccessCommand
import zio.ZIO

trait ShortenedUrlUserEventsService {
  def saveUserAccess(command: SaveShortenedUrlUserAccessCommand): ZIO[Any, Throwable, Unit]
}

File: myurlshortner-consumer/src/main/kotlin/com/acme/myurlshortner/consumer/domain/userevent/service/ShortenedUrlUserEventsService.kt
package com.acme.myurlshortner.consumer.domain.userevent.service

import com.acme.myurlshortner.consumer.domain.userevent.command.UserAccessedShortenedUrlCommand

interface ShortenedUrlUserEventsService {
    fun handleShortenedUrlUserAccessed(
        command: UserAccessedShortenedUrlCommand,
    )
}
